lib/alchemy/pg_search/page_extension.rb

```diff
@@ -1,54 +1,25 @@
 # Enable Postgresql full text indexing.
 #
-module Alchemy::PgSearch::PageExtension
-  def self.extended(base)
-    base.include InstanceMethods
-    base.include PgSearch::Model
+Alchemy::Page.class_eval do
+  include PgSearch::Model
 
-    base.pg_search_scope(
-      :full_text_search,
-      against: {
-        meta_description: "B",
-        meta_keywords: "B",
-        title: "B",
-        name: "A",
-      },
-      associated_against: {
-        searchable_essence_texts: :body,
-        searchable_essence_richtexts: :stripped_body,
-        searchable_essence_pictures: :caption,
-      },
-      using: {
-        tsearch: { prefix: true },
-      },
-    )
+  after_save :remove_unpublished_pages
 
-    base.has_many(
-      :searchable_contents,
-      -> {
-        where(alchemy_elements: { public: true })
-          .where(searchable: true)
-      },
-      class_name: "Alchemy::Content",
-      through: :all_elements,
-    )
+  multisearchable(
+    against: [
+      :meta_description,
+      :meta_keywords,
+      :name,
+    ],
+    additional_attributes: -> (page) { { page_id: page.id } },
+    if: :searchable?
+  )
 
-    Alchemy::PgSearch::SEARCHABLE_ESSENCES.each do |klass|
-      base.has_many(
-        :"searchable_#{klass.underscore.pluralize}",
-        class_name: "Alchemy::#{klass}",
-        source_type: "Alchemy::#{klass}",
-        through: :searchable_contents,
-        source: :essence,
-      )
-    end
+  def remove_unpublished_pages
+    Alchemy::PgSearch::Search.remove_page self unless searchable?
```

Comment:
Kudos an den Ruby parser, dass der das auch ohne Klammern schnallt

```suggestion
    Alchemy::PgSearch::Search.remove_page(self) unless searchable?
```